util/ubuntu-setup

```diff
@@ -92,11 +123,40 @@ if $YES; then
     ARGS="-y"
 fi
 
-PACKAGES="$(sed 's/#.*//;/^$/d' $BASE/util/ubuntu-packages.txt)"
+TOOLCHAIN_PACKAGES="libnewlib-dev libnewlib-arm-none-eabi gcc-arm-none-eabi libstdc++-arm-none-eabi-newlib"
+if [ "ubuntu-18.04" = "$SYSTEM" ]; then
+	# check for missing compiler
+	ARM_COMPILER=true
+	if [[ $(apt-cache policy gcc-arm-none-eabi) = *"Installed: (none)"* ]]; then
+		ARM_COMPILER=false
+	else
+		echo "-- found existing ARM compiler install, verifying"
+	fi
+
+	# compiler is installed, check if the version is correct
+	if [ "$ARM_COMPILER" = true ]; then
+		if [[ $(apt-cache policy gcc-arm-none-eabi) = *"Installed: 15:4.9.3"* ]]; then
+			echo "-- detected good arm compiler version"
+		else
+			echo "-- detected bad arm compiler verison, purging"
+			apt-get purge $TOOLCHAIN_PACKAGES $ARGS
```

Comment:
Just because I made the above comment, this variable should *not* be quoted since we're counting on the spaces being evaluated. So this is good :)